src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useRef, useState } from 'react';
+import axiosBaseURL from '../../apis/axiosBaseURL';
+import InputTextArea from '../common/InputTextarea/InputTextarea';
+import ButtonBox from '../common/ButtonBox/ButtonBox';
+import messageIcon from '../../images/icons/messages.svg';
+import closeIcon from '../../images/icons/close.svg';
+import styles from './Modal.module.css';
+
+export default function Modal({ setModalOpen, id, name, image }) {
+  const [content, setContent] = useState('');
+
+  const handleTextareaChange = (e) => {
+    setContent(e.target.value);
+  };
+
+  // 모달 닫기 버튼(x버튼)을 누르면 모달이 닫힘.
+  const closeModal = () => {
+    setModalOpen(false);
+  };
+
+  //처음에는 모달이 열려있지 않기때문에 모달창의 상태를 감지하는 useRef를 null 상태로 설정
+  const modalRef = useRef(null);
+
+  // useRef가 감지하고 있는 모달창이 열려있고 &  클릭한 곳이 모달창이 아니라면 모달이 닫힘
+  useEffect(() => {
+    const handler = (event) => {
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        setModalOpen(false);
+      }
+    };
+
+    // mousedown때마다 handler 호출
+    document.addEventListener('mousedown', handler);
+
+    // 컴포넌트 언마운트시(화면에서 사라졌을 때) 이벤트 제거
+    return () => {
+      document.removeEventListener('mousedown', handler);
+    };
+  }, []);
+
+  const postQuestion = async () => {
+    const res = await axiosBaseURL.post(
+      `subjects/${id}/questions/`,
+      { content: content },
+      { headers: { 'Content-Type': 'application/json' } }
+    );
+    console.log(res);
+  };
+
+  const handleFormSubmit = async (e) => {
+    await postQuestion();
+    setContent('');
+  };
+
```

Comment:
```suggestion
  const handleFormSubmit= async (event) => {
    event.preventDefault();
    try {
      const res = await axiosBaseURL.post(
        `subjects/${id}/questions/`,
        { content: content },
        { headers: { 'Content-Type': 'application/json' } }
      );
      setContent('');
    } catch {
      console.error('에러가 발생했습니다!')
    }
};
```
form과 button에 전달되는 2개의 함수를 
이렇게 하나로 합치고 button의 `onClick={postQuestion}` 부분을 `type='submit'`으로 수정하면 좋을 거 같습니다

Reply:
적용했습니다!